src/service/resource.js

```diff
@@ -357,10 +365,18 @@ export class Resource {
     const viewport = this.resources_.getViewport();
     this.layoutBox_ = box;
 
+    const win = this.resources_.win;
+    const computedStyle = win./*OK*/getComputedStyle(this.element);
```

Comment:
I've done basically what you suggest. I've passed the existing and the new margins along, since we need the existing margins in mutateWork_() to know where the bottom displaced boundary is. So doesn't make much sense to pass a diff around just to turn it back into the original value later on.